pkg/comp-functions/functions/vshnpostgres/maintenance.go

```diff
@@ -178,3 +184,21 @@ func createMaintenanceSecret(instanceNamespace, sgNamespace, resourceName string
 		},
 	}
 }
+
+func setPsqlMinorVersion(svc *runtime.ServiceRuntime, desiredCluster *stackgresv1.SGCluster) error {
+	observedCluster := &stackgresv1.SGCluster{}
+	err := svc.GetObservedKubeObject(observedCluster, "cluster")
+	if errors.IsNotFound(err) {
+		// Cluster doesn't exist yet. So let's ignore it here
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("cannot get observed cluster object: %w", err)
+	}
+
+	pgVersion := observedCluster.Spec.Postgres.Version
```

Comment:
shouldn't you take from the status?

Reply:
The problem comes from trying to change this field, so it makes sense that we take this value.